storage/remote/queue_manager.go

```diff
@@ -929,6 +985,28 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue chan sample) {
 			s.droppedOnHardShutdown.Add(uint32(droppedSamples))
 			return
 
+		case <-s.reshard:
+			for {
+				sample, ok := <-queue
+				// Accept those samples that might have entered the queue between sending the resharding
+				// signal and blocking of incoming samples at the time of creation of newer queues.
+				if !ok {
+					break
+				}
+				pendingSamples[nPending].Labels = labelsToLabelsProto(sample.labels, pendingSamples[nPending].Labels)
```

Comment:
I don't think you can just keep adding to `pendingSamples` like this. The slice has a size equal to the max samples per send, which is almost always smaller than the capacity the `queue` channel. I think it would be necessary either to batch up pendingSamples, or just send the samples directly without converting them to pending and back.